main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -348,9 +363,15 @@ class Flix {
   }
 
   /**
-    * Returns the reachable root definitions.
+    * Marks the given source `src` as changed.
     */
-  def getReachableRoots: Set[Symbol.DefnSym] = reachableRoots.toSet
+  def markChanged(src: Ast.Source): Flix = {
+    if (src == null)
```

Comment:
Is this a realistic concern? Who's throwing `null`s around?

Reply:
We check for null in all public `Flix` methods since this is the main entry point to the compiler. (If someone wants to build a program that wraps Flix).